### ISSUE TITLE
Should the profiles shipped with RT contain gamut mappings?

### ISSUE THREAD

**Issue by Beep6581** [Beep6581/RawTherapee#3118]
This is a question mainly for @Desmis though of course everyone is free to chime in.

I am familiar only with LUT-type color profiles and how they relate to rendering intent, but not with the other types. As far as I know, the RT_* profiles we ship are LUT profiles, but they seem to lack gamut mapping tables. As a result, currently if you save your photo using ProPhoto as the working space and RT_sRGB as the output space, using "perceptual" or "relative colorimetric" rendering intent for the output profile will make no difference. Should we not include gamut mappings in them?

**Comment by Desmis** [comment]
RT_*profiles I build, are "ICC ver 2", and then I think they cannot include "gamut mappings"

We must build ICC V4...and I don't know how to do ?

I know one output profile ICC v4, we can fond on "ICC international consortium" eg: "sRGB_V4_ICC_preference"
With these profiles intent works !

**Comment by Beep6581** [comment]
http://www.argyllcms.com/doc/iccgamutmapping.html

I think I can add the gamut mappings to our RT_* profiles - I did that for my monitor profile already. I'll do that soon and upload them here for you to double-check.

**Comment by Desmis** [comment]
No problem :)

**Comment by atorger** [comment]
I don't think we should use gamut mapping tables in the output profile. I don't think anyone else is doing it. If I store a file in sRGB I expect the samples in the file to be sRGB and not ProPhoto or something else mapped to sRGB via the profile. With 8 bit files it's not so good use of bits either.

Static gamut mappings is pretty "primitive", coming from a time when computers did not have much computing power. Today I think most expect to use the raw converter color tools to make the colors fit (or not fit) into the target gamut. If we should have gamut mapping feature I think we should have a specific tool for it instead so the gamut mapping is complete when writing the file.

Almost all (commercial) camera profiles have gamut mapping in them though, to avoid outputting too saturated colors which become difficult to work with. This is an old tradition and also stems from the fact that the raw converters do not really have gamut compression tools. I think a modern design would not have gamut mapping in camera profiles, and not in output profile, but you would have an easy way to control gamut in the raw converter itself.

**Comment by Beep6581** [comment]
@atorger I was under the impression that including gamut mapping into an output profile lets one use RT to map colors from the working space (say ProPhoto) into the output space (say RT_sRGB) in such a way that if I was to strip the saved JPG from its profile, I would see sRGB colors, not ProPhoto colors. In other words, I thought the mapping lets RT do the conversion which is then permanently baked into the resulting image. Am I wrong?

**Comment by atorger** [comment]
Note that I'm a bit rusty on these things, I've only worked with it in the context of printing (I did an Argyll printing guide a while back http://www.ludd.ltu.se/~torger/photography/argyll-print.html)

I would need to test to figure out if it works as you say or not. We can surely in the RT pipeline choose to pre-process the samples and write the final result into the image, but I *think* that you then must strip the intent tables if you embed the profile, otherwise a color managed image viewer will run the samples again through the table.

I do see some value in being able to gamut map from say ProPhoto to sRGB, but I don't think it's the best way to do it by hard-coding a ProPhoto-to-sRGB gamut mapping table in the profile. A better way would be to implement a dynamic gamut-mapping tool that could take into account what colors that actually are in the picture. A static gamut mapping table will compress all colors, and as ProPhoto and sRGB differs quite much it will possibly compress a fair bit into the gamut so you will notice compression also of normal colors. Say if you have reds and blues in the picture, but only the reds are outside sRGB a static gamut compression table will probably compress those blues too even if it doesn't have to.

Most raw converters have some gamut compression in the camera profiles themselves, and then work with relative colorimetric with soft clipping. The idea is that if you have too saturated colors you use the color sliders to reduce saturation, otherwise you satisfy with the soft rolloff. Perceptual content only is discussed when printing (I *think*) and then it's only used sometimes as relative colorimetric does the job often better.

For printing you use the rendering intents of the printer profile. One problem here is that the perceptual table in the printer profile must assume what the source gamut is, I suppose many assume it's AdobeRGB (as cameras can write AdobeRGB jpegs, and wide gamut screens matches AdobeRGB quite well). Perceptual output is not so popular though, you typically use relative colorimetric and then use a color management engine with blackpoint compensation and "soft clipping" which makes the relative colorimetric intent behave much like a perceptual content but without the problem of static assumption of source gamut, as relative colorimetric has no gamut compression.

To softproof you would first convert forward through the printer profile (vision/colorimetric color to printer values, with desired intent), and then backward again with relative colorimetric (printer values to vision color) and then output through the monitor profile (vision color to monitor RGB values, with desired intent).

Perceptual intent is a bit broken due to the fact that the profile must assume the source gamut, there's no tag in the profile which says which source gamut it assumes, and you can't have more than one perceptual table. I think that's why relative colorimetric with "perceptual tricks" (BPC and rolloff/soft clipping) is generally more popular.

**Comment by adamreichold** [comment]
Sorry for being a bit late to all this, but from how I understand the discussion, adding the mappings into the profiles shipped with RT would serve exactly one purpose: Allowing people to use say `RT_sRGB` as an output profile with perceptual intent using a fixed (but not formally specified) working profile, e.g. `RT_ProPhoto`. But also getting incorrect results when using a different working profile.

If that understanding is correct, I would also argue against including the mappings as it is easy to abuse for a user who does not have all this information at hand. And users who do, can probably create exactly those profiles resp. profile combinations they are looking for.

(I also agree that the application should make it as easy and efficient as possible to actually produce a final image with a gamut that fits the output profile, but I think this should be tackled separately.)
